### ISSUE TITLE
nameless logger in /loggers when using log4j2

### ISSUE THREAD

**Issue by joshiste** [spring-projects/spring-boot#7372]
When doing a `GET` on the new `/loggers` endpoint from boot 1.5.x and using log4j2 as logger backend an unnamed logger (root) is listed.
Setting the loglevel for this logger is impossible via `POST` to `/loggers/` - it results in a `405 - Method not allowed`

```json
{
	"": {
		"configuredLevel": "INFO",
		"effectiveLevel": "INFO"
	},
	"org.apache.catalina.startup.DigesterFactory": {
		"configuredLevel": "ERROR",
		"effectiveLevel": "ERROR"
	},
	"org.apache.catalina.util.LifecycleBase": {
		"configuredLevel": "ERROR",
		"effectiveLevel": "ERROR"
	},
	"org.apache.coyote.http11.Http11NioProtocol": {
		"configuredLevel": "WARN",
		"effectiveLevel": "WARN"
	},
	"org.apache.sshd.common.util.SecurityUtils": {
		"configuredLevel": "WARN",
		"effectiveLevel": "WARN"
	},
	"org.apache.tomcat.util.net.NioSelectorPool": {
		"configuredLevel": "WARN",
		"effectiveLevel": "WARN"
	},
	"org.crsh.plugin": {
		"configuredLevel": "WARN",
		"effectiveLevel": "WARN"
	},
	"org.crsh.ssh": {
		"configuredLevel": "WARN",
		"effectiveLevel": "WARN"
	},
	"org.eclipse.jetty.util.component.AbstractLifeCycle": {
		"configuredLevel": "ERROR",
		"effectiveLevel": "ERROR"
	},
	"org.hibernate.validator.internal.util.Version": {
		"configuredLevel": "WARN",
		"effectiveLevel": "WARN"
	},
	"org.springframework.boot.actuate.autoconfigure.CrshAutoConfiguration": {
		"configuredLevel": "WARN",
		"effectiveLevel": "WARN"
	},
	"org.springframework.boot.actuate.endpoint.jmx": {
		"configuredLevel": "WARN",
		"effectiveLevel": "WARN"
	},
	"org.thymeleaf": {
		"configuredLevel": "WARN",
		"effectiveLevel": "WARN"
	}
}
```

**Comment by wilkinsona** [comment]
For levels configured via `logging.level` in the environment, `LoggingApplicationListener` coerces `"root"` (case-insensitive) to `null`. We need to do something similar here.

**Comment by philwebb** [comment]
That empty logger is also to be addressed with #7386

**Comment by philwebb** [comment]
Closing in favor of PR #7372
